Web/src/components/RoomContainer/index.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {
+  useState, useEffect, EffectCallback, ReactElement,
+} from 'react';
+import { Socket, Channel } from 'phoenix';
+
+import getClientId from '../../utils/getClientId';
+import ChannelResponse from './ChannelResponse';
+
+const SOCKET_URL = 'ws://localhost:4000/socket';
+const TIMEOUT_DURATION = 5000;
+
+export enum ErrorCause {
+  Timeout = 'timeout',
+  Other = 'other'
+}
+
+interface RoomProps {
+  roomId: string,
+  render: (channel: Channel | null, error: [ErrorCause, any] | null) => ReactElement,
+}
+
+const RoomContainer: React.FC<RoomProps> = (props: RoomProps) => {
+  const [socket, setSocket] = useState<Socket | null>(null);
+  const [error, setError] = useState<[ErrorCause, any] | null>(null);
+  const [channel, setChannel] = useState<Channel | null>(null);
+
+  const { roomId, render } = props;
+
+  const maybeConnectSocket = () => {
```

Comment:
think should be fine, because the issue only arose because he use calling a function that was bound to the previous state - whereas we are returning the new state and using that as the argument for our next function.